tests/test_lane.py

```diff
@@ -1948,295 +1731,736 @@ def __contains__(self, k):
+        # This DatabaseBackedWorkList contains the YA book because its
+        # age range overlaps the age range of the book.
+        worklist_has_books(
+            [fourteen_or_fifteen], target_age=(12, 14)
+        )
 
-        gutenberg_adult = self._work(
+        worklist_has_books(
+            [adult, fourteen_or_fifteen],
+            audiences=[Classifier.AUDIENCE_ADULT], target_age=(12, 14)
+        )
+
+        # This lane contains no books because it skews too old for the YA
+        # book, but books for adults are not allowed.
+        older_ya = self._lane()
+        older_ya.target_age = (16,17)
+        self.add_to_materialized_view([older_ya])
+        worklist_has_books([], target_age=(16,17))
+
+        # Expand it to include books for adults, and the adult book
+        # shows up despite having no target age at all.
+        worklist_has_books([adult], target_age=(16, 18))
+
+    def test_audience_filter_clauses(self):
+        # Verify that audience_filter_clauses restricts a query to
+        # reflect a DatabaseBackedWorkList's audience filter.
+
+        # Create a children's book and a book for adults.
+        adult = self._work(
             title="Diseases of the Horse",
             with_license_pool=True, with_open_access_download=True,
             audience=Classifier.AUDIENCE_ADULT
         )
 
-        edition, lp = self._edition(
+        children = self._work(
             title="Wholesome Nursery Rhymes For All Children",
-            data_source_name=DataSource.OVERDRIVE,
-            with_license_pool=True
-        )
-        non_gutenberg_children = self._work(
-            presentation_edition=edition, audience=Classifier.AUDIENCE_CHILDREN
-        )
-        self.add_to_materialized_view(
-            [gutenberg_children, non_gutenberg_children, gutenberg_adult]
+            with_license_pool=True, with_open_access_download=True,
+            audience=Classifier.AUDIENCE_CHILDREN
         )
 
         def for_audiences(*audiences):
-            """Invoke WorkList.apply_audience_clauses using the given
+            """Invoke audience_filter_clauses using the given
             `audiences`, and return all the matching Work objects.
             """
-            wl = WorkList()
+            wl = DatabaseBackedWorkList()
             wl.audiences = audiences
-            qu = self._db.query(work_model).join(work_model.license_pool)
+            qu = wl.base_query(self._db)
             clauses = wl.audience_filter_clauses(self._db, qu)
             if clauses:
                 qu = qu.filter(and_(*clauses))
-            return [x.works_id for x in qu.all()]
-
-        eq_([gutenberg_adult.id], for_audiences(Classifier.AUDIENCE_ADULT))
+            return qu.all()
 
-        # The Gutenberg "children's" book is filtered out because it we have
-        # no guarantee it is actually suitable for children.
-        eq_([non_gutenberg_children.id],
-            for_audiences(Classifier.AUDIENCE_CHILDREN))
```

Comment:
This restriction has been removed. I've removed nearly all the Project Gutenberg books from the NYPL collection, so inappropriate "children's" books shouldn't be an issue anymore.